src/core/thread/mesh_forwarder.hpp

```diff
@@ -64,6 +64,103 @@ enum
  * @{
  */
 
+/**
+ * This class reprents an IPv6 fragment priority entry
+ *
+ */
+class FragmentPriorityEntry
+{
+public:
+    /**
+     * This method returns the fragment datagram tag value.
+     *
+     * @returns The fragment datagram tag value.
+     *
+     */
+    uint16_t GetDatagramTag(void) const { return mDatagramTag; }
+
+    /**
+     * This method sets the fragment datagram tag value.
+     *
+     * @param[in]  aDatagramTag  The fragment datagram tag value.
+     *
+     */
+    void SetDatagramTag(uint16_t aDatagramTag) { mDatagramTag = aDatagramTag; }
+
+    /**
+     * This method returns the source Rloc16 of the fragment.
+     *
+     * @returns The source Rloc16 value.
+     *
+     */
+    uint16_t GetSrcRloc16(void) const { return mSrcRloc16; }
+
+    /**
+     * This method sets the source Rloc16 value of the fragment.
+     *
+     * @param[in]  aSrcRloc16  The source Rloc16 value.
+     *
+     */
+    void SetSrcRloc16(uint16_t aSrcRloc16) { mSrcRloc16 = aSrcRloc16; }
+
+    /**
+     * This method returns the fragment priority value.
+     *
+     * @returns The fragment priority value.
+     *
+     */
+    uint8_t GetPriority(void) const { return mPriority; }
+
+    /**
+     * This method sets the fragment priotity value.
+     *
+     * @param[in]  aPriority  The fragment priority value.
+     *
+     */
+    void SetPriority(uint8_t aPriority) { mPriority = aPriority; }
+
+    /**
+     * This method returns the fragment priority entry's remaining lifetime.
+     *
+     * @returns The fragment priority entry's remaining lifetime.
+     *
+     */
+    uint8_t GetLifetime(void) const { return mLifetime; }
+
+    /**
+     * This method sets the remaining lifetime of the fragment priority entry.
+     *
+     * @param[in]  aLifetime  The remaining lifetime of the fragment priority entry (in seconds).
+     *
+     */
+    void SetLifetime(uint8_t aLifetime)
+    {
+        if (aLifetime > kMaxLifeTime)
+        {
+            aLifetime = kMaxLifeTime;
+        }
+
+        mLifetime = aLifetime;
+    }
+
+    /**
+     * This method decrements the entry lifetime.
+     *
+     */
+    void DecrementLifetime(void) { mLifetime--; }
+
+private:
+    enum
+    {
+        kMaxLifeTime = 31, ///< The maximum lifetime of the fragment entry (in seconds).
```

Comment:
Suggest reducing this value given that we are now resetting the lifetime on each received fragment. How about something like 5 seconds?

Reply:
Updated.